### ISSUE TITLE
Use type inside an `Object` in polymorhism

### ISSUE THREAD

**Issue by ronjunevaldoz** [Kotlin/kotlinx.serialization#1107]
**What is your use-case and why do you need this feature?**
Use type inside an `Object`

![image](https://user-images.githubusercontent.com/4947998/94577911-aa9bd500-02a9-11eb-8e82-576fb0809012.png)

**Describe the solution you'd like**

Base class
```
@Polymorphic
@Serializable()
abstract class BaseContent {
  @SerialName("_meta")
  abstract val meta: Meta
}
```
Sub class
``` 
@Serializable()
class PageContent {
  @SerialName("_meta")
  abstract val meta: Meta
} : BaseContent("https://cebu.onecms.content.page")
```

**Comment by qwwdfsad** [comment]
Could you please elaborate on what exactly you do want?
It's a bit unclear from the description of the issue

**Comment by ronjunevaldoz** [comment]
> Could you please elaborate on what exactly you do want?
> It's a bit unclear from the description of the issue

Hi, I have updated the details above. I am not sure if this feature is possible or if there is a better existing solution, then I would like that to know.

**Comment by sandwwraith** [comment]
If I got this correctly, you want the type discriminator (that `"type":"com.foobar.Class"` in Json) to be an actual object? That's not is not possible out-of-the-box; you can look into `JsonContentPolymorphicSerializer` and see if it helps you.

**Comment by ronjunevaldoz** [comment]
> If I got this correctly, you want the type discriminator (that `"type":"com.foobar.Class"` in Json) to be an actual object? That's not is not possible out-of-the-box; you can look into `JsonContentPolymorphicSerializer` and see if it helps you.

Yes, or if possible how do I point it just for the "schema" field. I will look at JsonContentPolymorphicSerializer
